libraries/Adafruit_LittleFS/src/Adafruit_LittleFS_File.cpp

```diff
@@ -272,6 +272,32 @@ uint32_t File::size (void)
   return ret;
 }
 
+bool File::truncate (uint32_t pos)
+{
+  uint32_t ret=0;
+  _fs->_lockFS();
+  if (!this->_is_dir)
```

Comment:
this seems to return true when the FIle is directory, I am not sure about that semantic, should we return false with directory !!!

Reply:
Yes, upon thought, truncate on a directory should return false. Pull updated.

Reply:
please use the `LFS_ERR_ISDIR` for error code with directory

Reply:
I'm not sure, change type to size_t and return error code or new size of file?
The other methods seem to avoid exposing LFS error codes and return 0 or false for inappropriate operations performed on a directory.

Reply:
@pyro9 I am not mentioning the returned code, I mean changing default `uint32_t ret=1;` into `int ret=LFS_ERR_ISDIR`;

Reply:
Now I understand. I have made the change.